CONTRIBUTING.md

```diff
@@ -1,3 +1,540 @@
+    - [Using pyenv](#using-pyenv)
+  - [Virtual Environments](#virtual-environments)
+    - [What is a Virtual Environment](#what-is-a-virtual-environment)
+      - [What about Python Package Management](#what-about-python-package-management)
+  - [Pipenv](#pipenv)
+    - [Common Issues](#common-issues)
+  - [Git Repo](#git-repo)
+    - [Forking SyMPC](#forking-sympc)
+    - [Clone GitHub Repo](#clone-github-repo)
+    - [Branching](#branching)
+    - [Syncing your Fork](#syncing-your-fork)
+    - [Learn More Git](#learn-more-git)
+  - [Setting up the VirtualEnv](#setting-up-the-virtualenv)
+    - [Pipenv](#pipenv-1)
+    - [Install Python Dependencies](#install-python-dependencies)
+    - [Linking the SyMPC src](#linking-the-sympc-src)
+  - [Code Quality](#code-quality)
+    - [Formatting and Linting](#formatting-and-linting)
+    - [Tests and CI](#tests-and-ci)
+    - [Writing Test Cases](#writing-test-cases)
+    - [Documentation and Code Style Guide](#documentation-and-code-style-guide)
+    - [Imports Formatting](#imports-formatting)
+    - [Generating Documentation](#generating-documentation)
+    - [Common Security Issues](#common-security-issues)
+  - [Pre-Commit](#pre-commit)
+    - [MacOS](#macos-3)
+    - [Creating a Pull Request](#creating-a-pull-request)
+    - [Check CI and Wait for Reviews](#check-ci-and-wait-for-reviews)
+  - [Support](#support)
+
+## Issues / PRs
+
+- Development is done on the `main` branch.
+- PR's may be closed without warning for any of the following reasons: lack of an associated Github issue, lack of tests, lack of proper documentation, or anything that isn't within the intended development roadmap of the Syft core team.
+- If you are working on an existing issue posted by someone else, please ask to be added as Assignee so that effort is not duplicated.
+- If you want to contribute to an issue someone else is already working on please get in contact with that person via slack or GitHub and discuss your collaboration.
+- If you wish to create your own issue or PR please explain your reasoning within the Issue template and make sure your code passes all the CI checks.
+
+**Caution**: We try our best to keep the assignee up-to-date, but as we are all humans with our own schedules mistakes happen. If you are unsure, please check the comments of the issue to see if someone else has already started work before you begin.
+
+### Beginner Issues
+
+If you are new to the project and want to get into the code, we recommend picking an issue with the label "good first issue". These issues should only require general programming knowledge and little to none insights into the project.
+
+## Requirements
+
+Before you get started you will need a few things installed depending on your operating system.
+
+- OS Package Manager
+- Python 3.7+
+- git
+
+### OSes
+
+We intend to provide first class support for dev setup in the current versions of:
+
+- 🐧 Ubuntu
```

Comment:
NITPICK: Leave only Linux :D (could you update the examples to remove the "brew")
We use `torchcsprng` that (I think) it is only available on Linux)

Reply:
I looked at the docs,``` torchcsprng ``` supports Windows(version>=0.2.0) .It supports mac only on CPU but not on CUDA. https://pypi.org/project/torchcsprng/

Reply:
Maybe, I should remove those parts and tell that torchspring can be use in windows what currentyle we do not provide any kind of support?

Reply:
Hey @rasswanth-s, Mac does not support CUDA so even Pytorch and torchcspring does'nt support cuda on Mac.

Reply:
Then I think, it would work on all three platforms.